src/lib.rs

```diff
@@ -259,13 +259,22 @@ impl<K: EnrKey> Enr<K> {
 
     /// Reads a custom key from the record if it exists.
     pub fn get(&self, key: impl AsRef<[u8]>) -> Option<&[u8]> {
+        // It's ok to decode any valid RLP integer as data
         self.get_raw_rlp(key).map(|rlp_data| {
             rlp::Rlp::new(rlp_data)
                 .data()
                 .expect("All data is sanitized")
         })
     }
 
+    // TODO @age: Should this fn be public?
+    fn get_integer<T: Decodable>(&self, key: impl AsRef<[u8]>) -> Option<T> {
```

Comment:
this works fine: `let _ : Option<String> = self.get_integer("key_name")` so it would be better to rename to 
```suggestion
    fn get_decodable<T: Decodable>(&self, key: impl AsRef<[u8]>) -> Option<T> {
```

Reply:
Adding this (`get_decodable` returning a `Result` as a public function I think it's actually quite a nice API improvement beyond the issue at hand